### DISCUSSION TITLE
Do I need to use JPH::BodyLockInterface when creating bodies

### DISCUSSION THREAD

**Discussion by bendgk** [jrouwe/JoltPhysics#885] [Q&A]
I'm currently writing a game engine using JoltPhysics and I am creating an abstraction over `CreateAndAddBody`, `AddBodiesPrepare`, and `AddBodiesFinalize`

My engine currently uses this architecture:
    * `World` class which manages physics scene
    * `World` has a `std::thread physics_thread` running at a fixed rate where `physics_system.Update()` is called (stepping)
    * `World` exposes an api for `create_and_add_body` and `create_and_add_bodies`

I'm wondering since my world steps/updates in a separate thread for my `create_and_add_body` and `create_and_add_bodies` methods do I need to use the locking version of `BodyInterface` using `physics_system.GetBodyLockInterface()`?

**Comment by jrouwe** [comment]
Yes, if you're simulating the world on one thread and calling CreateAndAddBody / AddBodiesPrepare or AddBodiesFinalize from another thread (and they can happen concurrently) you always need to use the locking version of the BodyInterface (PhysicsSystem::GetBodyInterface()). 

Note that during the PhysicsSystem::Update all bodies are locked, so CreateAndAddBody and AddBodiesFinalize will block until the Update finishes.
